### PR TITLE
Avoid crashing clients when fishing loot is empty, Fixes #4235

### LINKED ISSUE #4235: [BUG] Player is kicked while fishing
### What behaviour is expected:
Player can fish normally without being kicked

### What behaviour is observed:
Player is kicked randomly when fishing

Error Log:

```[07:15:54] [Server thread/ERROR]: Error whilst processing packet net.minecraft.server.v1_16_R2.PacketPlayInBlockPlace@134f697 for Thneed777[/108.39.147.160:53865]
java.lang.IllegalArgumentException: Cannot drop air
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.EntityItem.setItemStack(EntityItem.java:453) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.EntityItem.<init>(EntityItem.java:42) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.EntityFishingHook.b(EntityFishingHook.java:436) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.ItemFishingRod.a(ItemFishingRod.java:18) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.ItemStack.a(ItemStack.java:379) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.PlayerInteractManager.a(PlayerInteractManager.java:412) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.PlayerConnection.a(PlayerConnection.java:1521) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:28) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:5) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:23) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.TickTask.run(SourceFile:18) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.MinecraftServer.ba(MinecraftServer.java:1135) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.MinecraftServer.executeNext(MinecraftServer.java:1128) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.MinecraftServer.sleepForTick(MinecraftServer.java:1089) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:1003) ~[patched_1.16.2.jar:git-Paper-148]
	at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.2.jar:git-Paper-148]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

### Steps/models to reproduce:
1. Fishing
2. Crash randomly

### Plugin list:
Plugins (18): AcceptTheRules, BuycraftX, ChestSort, DeathLocation*, DeluxeMenus, DiscordSRV, Essentials, EssentialsChat, GSit, InventoryRollback, LuckPerms, PlaceholderAPI, ResourceWorld, SmoothSleep, spark, TAB, Vault, WorldEdit

All on the latest versions

### Paper version:
Paper version git-Paper-148 (MC: 1.16.2) (Implementing API version 1.16.2-R0.1-SNAPSHOT)

### Anything else:
We are using a custom datapack which modifies fishing loottables, but the tables are registered correctly and were working in 1.16.1; so we don't believe that is the source of the problem.

If the loot-table is required, I can post it as well.

### PR DESCRIPTION
**Pull request by schoentoon**

### CODE REVIEW

File: Spigot-Server-Patches/0594-Avoid-error-bubbling-up-when-item-stack-is-empty-in-.patch [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
++                    // Paper start, new EntityItem would throw if for whatever reason (mostly shitty datapacks) the itemstack1 turns out to be empty
++                    // if the item stack is empty we instead just have our entityitem as null
++                    EntityItem entityitem = null;
++                    if (!itemstack1.isEmpty()) {
++                        entityitem = new EntityItem(this.world, this.locX(), this.locY(), this.locZ(), itemstack1);
++                    }
++                    // Paper end
+                     // CraftBukkit start
+-                    PlayerFishEvent playerFishEvent = new PlayerFishEvent((Player) entityhuman.getBukkitEntity(), entityitem.getBukkitEntity(), (FishHook) this.getBukkitEntity(), PlayerFishEvent.State.CAUGHT_FISH);
++                    PlayerFishEvent playerFishEvent = new PlayerFishEvent((Player) entityhuman.getBukkitEntity(), entityitem != null ? entityitem.getBukkitEntity() : null, (FishHook) this.getBukkitEntity(), PlayerFishEvent.State.CAUGHT_FISH);
```
Comment (Proximyst) [inline Spigot-Server-Patches/0594-Avoid-error-bubbling-up-when-item-stack-is-empty-in-.patch]:
```suggestion
+                    PlayerFishEvent playerFishEvent = new PlayerFishEvent((Player) entityhuman.getBukkitEntity(), entityitem != null ? entityitem.getBukkitEntity() : null, (FishHook) this.getBukkitEntity(), PlayerFishEvent.State.CAUGHT_FISH); // Paper - entityitem may be null
```

File: Spigot-Server-Patches/0594-Don-t-crash-on-negative-item-count-for-fishing-loot.patch [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+diff --git a/src/main/java/net/minecraft/server/EntityFishingHook.java b/src/main/java/net/minecraft/server/EntityFishingHook.java
+index b6cace72ab2f3389408a5e528981fcff3b511180..f056192d513e27716136eaafe23d087e3fe42668 100644
+--- a/src/main/java/net/minecraft/server/EntityFishingHook.java
++++ b/src/main/java/net/minecraft/server/EntityFishingHook.java
+@@ -433,6 +433,7 @@ public class EntityFishingHook extends IProjectile {
+ 
+                 while (iterator.hasNext()) {
+                     ItemStack itemstack1 = (ItemStack) iterator.next();
++                    System.err.println("getCount() = " + itemstack1.getCount());
```
Comment (Proximyst) [inline Spigot-Server-Patches/0594-Don-t-crash-on-negative-item-count-for-fishing-loot.patch]: Debug

**Review by Proximyst — CHANGES_REQUESTED** [review]
You have to rebase onto master to have it not include aikar's commit.

**Review by Proximyst — APPROVED** [review]
LGTM. Thank you for your contribution.

**Review by aikar — APPROVED** [review]

**Pull request merged**